internal/kokoro/test.sh

```diff
@@ -24,10 +24,24 @@ git clone . $GAX_HOME
 cd $GAX_HOME
 
 try3() { eval "$*" || eval "$*" || eval "$*"; }
-try3 go get -v -t ./...
 
-./internal/kokoro/vet.sh
+download_deps() {
+    if [[ `go version` == *"go1.11"* ]] || [[ `go version` == *"go1.12"* ]]; then
+        export GO111MODULE=on
+        # All packages, including +build tools, are fetched.
+        try3 go mod download
+    else
+        # Because we don't provide -tags tools, the +build tools
+        # dependencies aren't fetched.
+        try3 go get -v -t ./...
+    fi
+}
+
+download_deps
 ./internal/kokoro/check_incompat_changes.sh
+./internal/kokoro/vet.sh
+go test -race -v . 2>&1 | tee $KOKORO_ARTIFACTS_DIR/$KOKORO_GERRIT_CHANGE_NUMBER.txt
 
-# Run tests and tee output to log file, to be pushed to GCS as artifact.
-go test -race -v ./... 2>&1 | tee $KOKORO_ARTIFACTS_DIR/$KOKORO_GERRIT_CHANGE_NUMBER.txt
+cd v2
```

Comment:
pushd, popd?

Reply:
> pushd, popd?

No need to come come back, so figured it's superfluous.

> why not go test -race -v ./...

Because we want to run `download_deps` and follow it up with `go test` independently in each module. `go test ./...` would run the v2 module's tests with the v1 module's dependencies. It would probably work in practice since their deps are largely the same, but I figure it could go subtly wrong.

Food for thought: this is going to be a reaaaaaal pain when we have modules several layers deep. Imagine a tree of root > some-dir > another-module. How do I get `go test ./...` to run recursively until it reaches another module? I dunno. I suspect we'll need another tool that finds all the directories of the existing module and pipes into `go test`, and a separate tool that finds all the modules in a repo.